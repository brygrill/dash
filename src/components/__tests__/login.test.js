import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import LoginComponent from '../login';

const myMock = jest.fn();

describe('<LoginComponent />', () => {
  it('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<LoginComponent login={myMock} error={false} />, div);
  });
  it('Fires login function on submit', () => {
    const wrapper = mount(<LoginComponent login={myMock} error={false} />);
    wrapper.find('Form').simulate('submit');
    expect(myMock.mock.calls.length).toBe(1);
  });
  it('Matches snapshot with no error', () => {
    const component = renderer.create(
      <LoginComponent login={myMock} error={false} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Matches snapshot with error', () => {
    const component = renderer.create(<LoginComponent login={myMock} error />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
