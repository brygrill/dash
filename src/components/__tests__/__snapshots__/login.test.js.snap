exports[`<LoginComponent /> Matches snapshot with error 1`] = `
<div
  className="ui centered relaxed sixteen column grid"
  style={
    Object {
      "alignContent": "center",
      "justifyContent": "center",
      "margin": 0,
      "width": "100%",
    }
  }>
  <div
    className="center aligned middle aligned sixteen wide column"
    style={
      Object {
        "maxWidth": undefined,
        "padding": 0,
      }
    }>
    <div
      className="blurring dimmable">
      <div
        className="ui segment ui inverted dimmer"
        onClick={[Function]}>
        <div
          className="content">
          <div
            className="center">
            <div
              className="ui loader" />
          </div>
        </div>
      </div>
      <div
        className="ui raised segment"
        style={
          Object {
            "margin": 0,
            "paddingTop": "2rem",
          }
        }>
        <img
          alt={undefined}
          className="ui small centered image"
          height={undefined}
          src={undefined}
          width={undefined} />
        <div
          className="ui medium header"
          style={
            Object {
              "color": "#ABB7B7",
              "fontWeight": "300",
              "marginTop": "1rem",
            }
          }>
          Login and Plan Your Training.
        </div>
        <div
          style={
            Object {
              "padding": "1rem",
            }
          }>
          <button
            className="ui large button"
            onClick={[Function]}
            style={
              Object {
                "backgroundStyle": "#d4d4d5",
              }
            }
            tabIndex={undefined}>
            <i
              aria-hidden="true"
              className="google icon" />
             
            Login with Google
          </button>
          <div
            className="ui horizontal divider"
            style={
              Object {
                "display": "none",
                "margin": 0,
                "paddingTop": "1rem",
              }
            }>
            Or
          </div>
        </div>
        <form
          className="ui large form"
          onSubmit={[Function]}
          style={
            Object {
              "display": "none",
            }
          }>
          <div
            className="ui basic segment">
            <div
              className="field">
              <div
                className="ui left icon input">
                <i
                  aria-hidden="true"
                  className="user icon" />
                <input
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  required={undefined}
                  type="email"
                  value="" />
              </div>
            </div>
            <div
              className="field">
              <div
                className="ui left icon input">
                <i
                  aria-hidden="true"
                  className="lock icon" />
                <input
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  required={undefined}
                  type="password"
                  value="" />
              </div>
            </div>
            <div
              className="field">
              <button
                className="ui large fluid primary button"
                onClick={[Function]}
                required={undefined}
                style={
                  Object {
                    "backgroundColor": undefined,
                  }
                }
                tabIndex={undefined}
                type={undefined}>
                LOGIN
              </button>
            </div>
          </div>
        </form>
        <div
          className="ui error message">
          <div
            className="content">
            <div
              className="header">
              Login Error
            </div>
            <p>
              Please enter valid credentials.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<LoginComponent /> Matches snapshot with no error 1`] = `
<div
  className="ui centered relaxed sixteen column grid"
  style={
    Object {
      "alignContent": "center",
      "justifyContent": "center",
      "margin": 0,
      "width": "100%",
    }
  }>
  <div
    className="center aligned middle aligned sixteen wide column"
    style={
      Object {
        "maxWidth": undefined,
        "padding": 0,
      }
    }>
    <div
      className="blurring dimmable">
      <div
        className="ui segment ui inverted dimmer"
        onClick={[Function]}>
        <div
          className="content">
          <div
            className="center">
            <div
              className="ui loader" />
          </div>
        </div>
      </div>
      <div
        className="ui raised segment"
        style={
          Object {
            "margin": 0,
            "paddingTop": "2rem",
          }
        }>
        <img
          alt={undefined}
          className="ui small centered image"
          height={undefined}
          src={undefined}
          width={undefined} />
        <div
          className="ui medium header"
          style={
            Object {
              "color": "#ABB7B7",
              "fontWeight": "300",
              "marginTop": "1rem",
            }
          }>
          Login and Plan Your Training.
        </div>
        <div
          style={
            Object {
              "padding": "1rem",
            }
          }>
          <button
            className="ui large button"
            onClick={[Function]}
            style={
              Object {
                "backgroundStyle": "#d4d4d5",
              }
            }
            tabIndex={undefined}>
            <i
              aria-hidden="true"
              className="google icon" />
             
            Login with Google
          </button>
          <div
            className="ui horizontal divider"
            style={
              Object {
                "display": "none",
                "margin": 0,
                "paddingTop": "1rem",
              }
            }>
            Or
          </div>
        </div>
        <form
          className="ui large form"
          onSubmit={[Function]}
          style={
            Object {
              "display": "none",
            }
          }>
          <div
            className="ui basic segment">
            <div
              className="field">
              <div
                className="ui left icon input">
                <i
                  aria-hidden="true"
                  className="user icon" />
                <input
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  required={undefined}
                  type="email"
                  value="" />
              </div>
            </div>
            <div
              className="field">
              <div
                className="ui left icon input">
                <i
                  aria-hidden="true"
                  className="lock icon" />
                <input
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  required={undefined}
                  type="password"
                  value="" />
              </div>
            </div>
            <div
              className="field">
              <button
                className="ui large fluid primary button"
                onClick={[Function]}
                required={undefined}
                style={
                  Object {
                    "backgroundColor": undefined,
                  }
                }
                tabIndex={undefined}
                type={undefined}>
                LOGIN
              </button>
            </div>
          </div>
        </form>
        <div
          className="ui error hidden message">
          <div
            className="content">
            <div
              className="header">
              Login Error
            </div>
            <p>
              Please enter valid credentials.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
